sktime/markov/msm/markov_state_model.py

```diff
@@ -1213,3 +1221,145 @@ def score(self, dtrajs, score_method='VAMP2', score_k=10):
         from sktime.metrics import vamp_score
         return vamp_score(K, C00_train, C0t_train, Ctt_train, C00_test, C0t_test, Ctt_test,
                           k=score_k, score=score_method)
+
+
+class MarkovStateModelCollection(MarkovStateModel):
+    r"""A collection of Markov state models. An instance of this itself behaves like a
+    :class:`MSM <MarkovStateModel>`, defaulting to the first (transition matrix, stationary distribution,
+    count model) triplet provided. The model can be switched by a call to :meth:`select`.
+
+    See Also
+    --------
+    MarkovStateModel : Markov state model.
+    MaximumLikelihoodMSM : Type of estimator that produces this model.
+    """
+
+    def __init__(self, transition_matrices: List[np.ndarray],
+                 stationary_distributions: List[np.ndarray],
+                 reversible: Optional[bool],
+                 count_models: List[TransitionCountModel],
+                 transition_matrix_tolerance: float):
+        r"""Creates a new collection of Markov state models.
+
+        Parameters
+        ----------
+        transition_matrices : list of (n, n) ndarray
+            List of transition matrices. Each of them must be row-stochastic.
+        stationary_distributions : list of `None` or list of (n, ) ndarray
+            List of stationary distributions belonging to transition matrices.
+            If list of None, it will be evaluated upon construction of the instance.
+        reversible : bool or None
+            Whether the transition matrices are reversible with respect to its stationary distribution.
+            If None, this is determined from the transition matrix.
+        count_models : List of count models or list of None
+            List of count models belonging to transition matrices.
+        transition_matrix_tolerance : float
+            Tolerance under which a matrix is considered to be a transition matrix.
+        """
+        n = len(transition_matrices)
+        if n == 0:
+            raise ValueError("Needs at least one transition matrix!")
```

Comment:
if this class can be called from another class, maybe make the error message clearer by mentioning the name of the class?

Reply:
Doesn't this appear in the stack trace anyways?

Reply:
yeah true, so not needed